### PR TITLE
Documentation Improvements

### PR DESCRIPTION
**Pull request by hedgerh**
Added some extra stuff to the README to clear up confusion that I and others had when trying to get middleware working in websockets.  This closes https://github.com/kudos/koa-websocket/issues/10

Also added a simple client example to close https://github.com/kudos/koa-websocket/issues/8 .

### CODE REVIEW

**Comment by kudos** [comment]
This is great, thanks! I'm about to board a super long flight but might be able to review on board if the wifi works.

**Comment by hedgerh** [comment]
Thank you too.  I couldnt figure out how to access my session context in websockets for the life of me and this did the trick.  Wanted to clear up some things in the docs that I had to figure out.

**Comment by hedgerh** [comment]
@chovy - we should probably discuss the changes in this PR.  i appreciate any feedback about what could be more clear.

I went with the `'/'` index route instead of something like `/foo` because i was kinda confused when i read the original docs use of `/tests/:id`.  I did have a little bit of trouble trying to clearly explain that part with just the index route, though.  I'll try and rework it to be more clear about whats going on.

File: README.md [hunk 1]
```diff
@@ -8,26 +8,108 @@
...
+
+## Accessing the `this.websocket` instance in the Koa app
+If you'd like to access the WebSocket instance from Koa using `this.websocket`, we need to add the `this.websocket` context to our Koa app from inside our WebSockets handler.
+
+```js
+app.ws.use(route.all('/', function* (next) {
+  // access the main Koa app's context via `app.context`.
+  // Here we add `this.websocket` from our WebSockets context to the main Koa app's context.
+  // It will then be available in Koa's context via `this.websocket`.
+  app.context.websocket = this.websocket;
```
Comment (kudos) [inline README.md]: This seems kind of wrong to me, but I'm very jetlagged.

Comment (kudos) [inline README.md]: @hedgerh thoughts on this?

**Comment by kudos** [comment]
@hedgerh I'm going to merge this and change the part I had a question on.

**Pull request merged**